packages/app/src/systems/Transaction/services/transaction.tsx

```diff
@@ -186,11 +187,81 @@ export class TxService {
       (account?.address?.toString() || address) as string,
       provider
     );
+    const abiMap = await getAbiMap({
+      inputs: transactionRequest.toTransaction().inputs,
+    });
+
+    const expectedSummary = await getTransactionSummaryFromRequest({
+      provider,
+      transactionRequest,
+      abiMap,
+    });
+
```

Comment:
this is never gonna go wrong as you are comparing the transactionSummary from the same txRequest twice. 

what needs to be done is to compare with the transactionSummary received from the SDK side (if we have it)